src/displayBackend/wayland/PresentationFeedbackSurface.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, write to the Free Software
+ *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
+ *
+ * Copyright (C) 2022 EPAM Systems Inc.
+ *
+ */
+
+#include "PresentationFeedbackSurface.hpp"
+#include "FrameBuffer.hpp"
+#include "Exception.hpp"
+#include "Compositor.hpp"
+#include "Display.hpp"
+
+using std::mutex;
+using std::thread;
+using std::unique_lock;
+
+using DisplayItf::FrameBufferPtr;
+
+namespace Wayland {
+
+PresentationFeedbackSurface::PresentationFeedbackSurface(wl_compositor* compositor)
+: Surface(compositor)
+{
+	mWlPresentationFeedbackListener = {
+		sPresentationFeedbackHandleSyncOutput, sPresentationFeedbackHandlePresented,
+		sPresentationFeedbackHandleDiscarded
+	};
+
+	auto wr = Display::getWaylandRegistry().getItem(wp_presentation_interface.name);
+	mWlPresentation = (wp_presentation*)wl_registry_bind(wr.registry, wr.id, &wp_presentation_interface, 1);
```

Comment:
We have many surfaces but registry should be bound only once. Better to bind/destroy it in `display` and pass wp_presentation* as parameter.

Reply:
Disagree with that. The proposed solution is to create a huge object(Display) with enormous dependencies ( you bind in display and need to pass it through to the correct place). The logic, in that case, will be split among several objects with all consequences regarding object destruction. The current solution accumulates all presentation logic in one object.